lib/message_queue.cpp

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2020 - 2022 Pionix GmbH and Contributors to EVerest
+#include <thread>
+
+#include <fmt/format.h>
+
+#include <everest/logging.hpp>
+
+#include <utils/message_queue.hpp>
+
+namespace Everest {
+
+Message::Message(const std::string& topic, const std::string& payload) : topic(topic), payload(payload) {
+}
+
+MessageQueue::MessageQueue(const std::function<void(std::shared_ptr<Message> message)>& message_callback) :
+    message_callback(message_callback), running(true) {
+    this->worker_thread = std::thread([this]() {
+        while (this->running) {
+            std::shared_ptr<Message> message;
+            std::unique_lock<std::mutex> lock(this->message_mutex);
+            this->cv.wait(lock, [this]() { return !this->message_queue.empty(); });
+
+            message = this->message_queue.front();
```

Comment:
you could use auto message = std::move(this->message_queue.front()) here, instead of initializing, copying and destroying it, but the pop should follow closeby